### PR TITLE
[REF] mail: several messaging code cleaning

### PR DESCRIPTION
**Pull request by alexkuhn**
- [REF] mail: pass model instead of id in Messaging methods
  TestServer tests are skipped because cannot trivially be converted from this change [REF] mail: pass thread instead of local id in component props [REF] mail: introduce setter ChatWindow/thread
- [REF] mail: pass thread instead of local id to useAttachmentUploader
  also use positional params
- [REF] mail: make template more compact

### CODE REVIEW

File: addons/mail/static/src/new/core/chat_window_model.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class ChatWindow {
      */
     constructor(state, data) {
         Object.assign(this, {
-            threadLocalId: data.threadLocalId,
+            thread: data.thread,
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/chat_window_model.js:69]: I'm not sure to understand on this model thread is both assigned as field and has a getter?

**Review by seb-odoo — COMMENTED** [review]
I like the idea, ok if everything is covered by tests.

I don't like that we skip more tests, especially during such change, I would prefer if the tests were adapted first.

File: addons/mail/static/src/new/core/chat_window_model.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/core/chat_window_model.js:69]: Yes, I defined `set thread`, so it invokes it here, which does `threadLocalId = thread.localId`.

**Comment by alexkuhn** [comment]
> I don't like that we skip more tests, especially during such change, I would prefer if the tests were adapted first.

At a glance, I think all the test I skipped have their coverage from other `startServer` tests.
I skipped them just to not delete them immediately, as to keep a trace whether we should translate to equivalent `startServer()` test or not.

Cleaning code is far more important than keeping these bad tests.

**Pull request merged**